tss-esapi/tests/context_tests/tpm_commands/duplication_commands_tests.rs

```diff
@@ -1,2 +1,57 @@
 // Copyright 2021 Contributors to the Parsec project.
 // SPDX-License-Identifier: Apache-2.0
+mod test_duplicate {
+    use crate::common::{create_ctx_with_session, decryption_key_pub};
+    use std::convert::TryFrom;
+    use tss_esapi::handles::ObjectHandle;
+    use tss_esapi::structures::SymmetricDefinitionObject;
+    use tss_esapi::{interface_types::resource_handles::Hierarchy, structures::Auth};
+
+    #[test]
+    fn test_duplicate() {
+        let mut context = create_ctx_with_session();
+
+        let random_digest = context.get_random(16).unwrap();
+        let key_auth = Auth::try_from(random_digest.value().to_vec()).unwrap();
+
+        let object_to_duplicate_handle: ObjectHandle = context
+            .create_primary(
+                Hierarchy::Owner,
+                &decryption_key_pub(),
+                Some(&key_auth),
+                None,
+                None,
+                None,
+            )
+            .unwrap()
+            .key_handle
+            .into();
+
+        let new_parent_handle: ObjectHandle = context
+            .create_primary(
+                Hierarchy::Owner,
+                &decryption_key_pub(),
+                Some(&key_auth),
+                None,
+                None,
+                None,
+            )
+            .unwrap()
+            .key_handle
+            .into();
+
+        context
+            .tr_set_auth(object_to_duplicate_handle, &key_auth)
+            .unwrap();
+        context.tr_set_auth(new_parent_handle, &key_auth).unwrap();
+
+        let _ = context
+            .duplicate(
+                object_to_duplicate_handle,
+                new_parent_handle,
+                None,
+                SymmetricDefinitionObject::Null,
+            )
+            .unwrap();
```

Comment:
I think you need to crate a policy session for this call and follow the steps mentioned in the spec for this command to set it up. Unfortunately, it's probably going to be somewhat trial-and-error.

Reply:
Okay, thanks for the hint! I already had a policy session that allowed duplication but then key creation failed. I'll check it out, thanks!

Reply:
I think that session should only be used for the duplicate call, not for creation as well!